src/EditorFeatures/CSharpTest/CodeActions/MoveType/MoveTypeTests.MoveToNewFile.cs

```diff
@@ -1275,6 +1275,101 @@ await TestMoveTypeToNewFileAsync(
             code, codeAfterMove, expectedDocumentName, destinationDocumentText);
     }
 
+    [WpfFact, WorkItem("https://github.com/dotnet/roslyn/issues/19613")]
+    public async Task MoveTypeWithDirectives3()
+    {
+        var code =
+            """
+            public class Goo
+            {
+                #region Region
+                public class [||]Bar
+                {
+                }
+                #endregion
+            }
+            """;
+        var codeAfterMove =
+            """
+            public partial class Goo
+            {
+
+                #region Region
+                #endregion
+            }
+            """;
+
+        var expectedDocumentName = "Bar.cs";
+        var destinationDocumentText =
+            """
+            public partial class Goo
+            {
+                #region Region
+                public class Bar
+                {
+                }
+                #endregion
+            }
+            """;
+
+        await TestMoveTypeToNewFileAsync(
+            code, codeAfterMove, expectedDocumentName, destinationDocumentText);
+    }
+
+    [WpfFact, WorkItem("https://github.com/dotnet/roslyn/issues/19613")]
+    public async Task MoveTypeWithDirectives4()
+    {
+        var code =
+            """
+            public class Goo
+            {
+                #region Region1
+                public class NotBar
+                {
+                }
+                #endregion
+
+                #region Region2
+                public class [||]Bar
+                {
+                }
+                #endregion
+            }
+            """;
+        var codeAfterMove =
+            """
+            public partial class Goo
+            {
+                #region Region1
+                public class NotBar
+                {
+                }
+
+                #endregion
+                #region Region2
+                #endregion
+            }
+            """;
+
+        var expectedDocumentName = "Bar.cs";
+        var destinationDocumentText =
+            """
+            public partial class Goo
+            {
+                #endregion
```

Comment:
I am a little confused, this `#endregion` directive doesn't look desired.

Reply:
oh yeah.  that seems odd.  looking.

Reply:
fixed up on IDE side.